packages/wallets/src/features/cashier/flows/WalletWithdrawal/WalletWithdrawal.tsx

```diff
@@ -1,27 +1,48 @@
 import React, { useEffect, useState } from 'react';
-import { WithdrawalFiatModule, WithdrawalVerificationModule } from '../../modules';
+import { useAuthorize, useCurrencyConfig } from '@deriv/api';
+import { WithdrawalCryptoModule, WithdrawalFiatModule, WithdrawalVerificationModule } from '../../modules';
+import { Loader } from '../../../../components/Loader';
 
 const WalletWithdrawal = () => {
+    const {
+        data: authorizeData,
+        isLoading: isAuthorizeLoading,
+        isSuccess: isAuthorizeSuccess,
+        switchAccount,
+    } = useAuthorize();
+    const { getConfig, isLoading: isCurrencyConfigLoading, isSuccess: isCurrencyConfigSuccess } = useCurrencyConfig();
     const [verificationCode, setVerificationCode] = useState('');
 
     useEffect(() => {
         const queryParams = new URLSearchParams(location.search);
         const verificationQueryParam = queryParams.get('verification');
+        const walletIdParam = queryParams.get('walletid');
 
-        if (verificationQueryParam) {
+        if (verificationQueryParam && walletIdParam) {
             setVerificationCode(verificationQueryParam);
+            switchAccount(walletIdParam); // authorize the intended wallet
 
             const url = new URL(window.location.href);
             url.searchParams.delete('verification'); // Remove the 'verification_code' query parameter
+            url.searchParams.delete('walletid'); // Remove the 'loginid' of the wallet from the query parameters
             window.history.replaceState({}, document.title, url.toString());
         }
     }, []);
```

Comment:
`switchAccount` should be added as a dependency for the useEffect

Reply:
@lubega-deriv. Sorry, couldn't catch your point.

Reply:
@aum-deriv shouldn't `switchAccount` be added as a dependency for the useEffect so that it would render when the function is called based on `walletIdParam`